lib/uklibparam/include/uk/libparam.h

```diff
@@ -187,18 +130,16 @@ UK_CTOR_PRIO(UK_LIBPARAM_LIBDESC_CTOR, 0);
 
 #define __UK_LIBPARAM_PARAM_DEFINE(arg_var, arg_addr, arg_type, arg_count, \
 				   arg_desc)				\
-	static struct uk_libparam_param __UK_LIBPARAM_PARAM_NAME(arg_var) = { \
-		.name  = STRINGIFY(arg_var),				\
-		.desc  = arg_desc,					\
-		.type  = _UK_LIBPARAM_PARAM_TYPE_##arg_type,		\
-		.count = arg_count,					\
-		.addr  = arg_addr					\
+	static const struct uk_libparam_param				\
```

Comment:
Hum... we are touching `__widx` later for parsing... should we really declare the struct as `const`? Doing later a DECONST feels wrong... In principle we could also move any `rw` fields to another static `struct` and reference to that one from the parameter struct. This extra struct does not need to be in a special section.

Reply:
You're right, the only reason I didn't do it is because it's only `_widx` that is non-const in that struct, so we would need to declare a struct just for that, similarly to:
```c
struct uk_libparam_data {
        /* Internally used by the parser for array parameters */
        __sz __widx;
};

#define __UK_LIBPARAM_PARAM_DEFINE(arg_var, arg_addr, arg_type, arg_count,
                                   arg_desc) \
        static const struct uk_libparam_data __UK_LIBPARAM_DATA_NAME(arg_var); \
        static const struct uk_libparam_param \
        __used __section("." STRINGIFY(UK_LIBPARAM_PARAMSECTION_NAME)) \
        __align(8) __UK_LIBPARAM_PARAM_NAME(arg_var) = { \
                .prefix = STRINGIFY(UK_LIBPARAM_LIBPREFIX), \
                .name   = STRINGIFY(arg_var), \
                .desc   = arg_desc, \
                .type   = _UK_LIBPARAM_PARAM_TYPE_##arg_type, \
                .count  = arg_count, \
                .addr   = arg_addr \
                .data   = &__UK_LIBPARAM_DATA_NAME(arg_var); \
        };
```
Is that what you essentially propose?

PS: Yes `data` is a very uninspired name.

Reply:
Yes, I think we should do this. This would clean-up the declarations. Should `const` be removed from `data`?